tests/test.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+set -e
+
+CURRENT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+TEST_DATA=${EPHEMERIS_TEST_DATA:-"$CURRENT_DIR"}
+
+run-data-managers --help
+shed-install --help
+workflow-install --help
+setup-data-libraries --help
+get-tool-list --help
+
+echo "Starting galaxy docker container"
+CID=`docker run -d -e GALAXY_CONFIG_WATCH_TOOL_DATA_DIR=True -p 8080:80 bgruening/galaxy-stable`
```

Comment:
This depends on the docker run executable to output the container name. This may change upstream. Therefore I prefer to use the --name flag.
Using a ENV variable for this is a good idea indeed.

Reply:
They are using this in their own documentation, and this is such a common part of the API that it would be really bad if they changed it. If you specify a name you're guaranteed to clash if the test has been running / if a user has a container of the same name running.

Reply:
You are right. Will change that back.

Reply:
Similarly we shouldn't expose any ports, and get the internal container's ip address to be more robust in that regard. Certainly 8080 is not a great idea.

Reply:
This can be fixed by only exposing it to localhost.

Reply:
That won't work on OSX. I'd prefer the `-P` option and the using inspect to grab the webport.
`docker inspect --format='{{(index (index .NetworkSettings.Ports "80/tcp") 0).HostPort}}' $CID` will give you the random port that been mapped to 80.

Reply:
Even better is to use the containers internal ip address to communicate with it. Prevents port collision all together.

Reply:
No, that doesn't work on OSX https://docs.docker.com/docker-for-mac/networking/#i-cannot-ping-my-containers

Reply:
Okay. I do not have access to a mac. so there is no way I can test whatever I write does work on OSX. Would you be so kind to implement this?